cellprofiler/__main__.py

```diff
@@ -69,7 +70,8 @@ def main(args=None):
 
     options, args = parse_args(args)
 
-    if any(options.pipeline_filename.startswith(protocol) for protocol in ('http', 'https', 'ftp')):
+    o = urlparse.urlparse(options.pipeline_filename)
+    if o[0] in ("ftp", "http", "https"):
         import urllib2
```

Comment:
Need to add something like the below before the `temp_pipe_file` assignment, or else an error can occur if the default temp location doesn't exist:
```
if options.temp_dir is not None:
    if not os.path.exists(options.temp_dir):
        os.makedirs(options.temp_dir)
        temp_dir = options.temp_dir
    else:
        temp_dir = cpprefs.get_temporary_directory()
```